src/main/java/org/opensearch/plugin/insights/core/exporter/LocalIndexExporter.java

```diff
@@ -113,4 +119,47 @@ public void close() {
     private String getDateTimeFromFormat() {
         return indexPattern.print(DateTime.now(DateTimeZone.UTC));
     }
+
+    /**
+     * Set local index exporter data retention period
+     *
+     * @param deleteAfter the number of days after which Top N local indices should be deleted
+     */
+    public void setDeleteAfter(final int deleteAfter) {
+        this.deleteAfter = deleteAfter;
+    }
+
+    /**
+     * Delete Top N local indices older than the configured data retention period
+     */
+    public void deleteExpiredIndices(final Map<String, IndexMetadata> indexMetadataMap) {
+        long expirationMillisLong = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(deleteAfter);
+        for (IndexMetadata indexMetadata : indexMetadataMap.values()) {
+            String indexName = indexMetadata.getIndex().getName();
+            if (!matchesPattern(indexName, indexPattern)) {
+                continue;
+            }
+            if (indexMetadata.getCreationDate() <= expirationMillisLong) {
+                // delete this index
+                client.admin().indices().delete(new DeleteIndexRequest(indexMetadata.getIndex().getName()));
+            }
+        }
+    }
+
+    /**
+     * Checks if the input string matches the given DateTimeFormatter pattern.
+     *
+     * @param input The input string to check.
+     * @param formatter The DateTimeFormatter to validate the string against.
+     * @return true if the string matches the pattern, false otherwise.
+     */
+    static boolean matchesPattern(final String input, final DateTimeFormatter formatter) {
+        try {
+            // Try parsing the input with the given formatter
+            formatter.parseDateTime(input);
+            return true;  // String matches the pattern
+        } catch (IllegalArgumentException e) {
```

Comment:
According to https://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormatter.html#parseDateTime-java.lang.String-, `parseDateTime` can also throw [UnsupportedOperationException](https://docs.oracle.com/javase/1.5.0/docs/api/java/lang/UnsupportedOperationException.html?is-external=true). Let's try to catch all the exceptions here.

Reply:
I will change it to this:
```
        try {
            // Try parsing the input with the given formatter
            formatter.parseDateTime(input);
            return true;  // String matches the pattern
        } catch (Exception e) {
            return false;  // String does not match the pattern
        }
```
We only care if the string matches so ignoring all exception types.